chain-signatures/node/src/rpc.rs

```diff
@@ -645,3 +691,76 @@ async fn try_publish_eth(
     }
     // TODO: add metrics for Ethereum
 }
+
+use chain_signatures_project::instruction::Respond as SolanaRespond;
+use chain_signatures_project::Signature as SolanaContractSignature;
+use chain_signatures_project::AffinePoint as SolanaContractAffinePoint;
+use chain_signatures_project::accounts::Respond as SolanaRespondAccount;
+use solana_sdk::signature::Signer as SolanaSigner;
+async fn try_publish_sol(
+    sol: &SolanaClient,
+    action: &PublishAction,
+    timestamp: &Instant,
+    signature: &Signature,
+) -> Result<(), ()> {
+    let program = sol.client.program(sol.program_id).map_err(|_| ())?;
+    
+    let request_ids = vec![action.request.indexed.id.request_id];
+    let signature = SolanaContractSignature {
+        big_r: SolanaContractAffinePoint {
+            x: signature.big_r.to_encoded_point(false).as_bytes()[1..33].try_into().unwrap(),
+            y: signature.big_r.to_encoded_point(false).as_bytes()[33..65].try_into().unwrap(),
+        },
+        s: signature.s.to_bytes().into(),
+        recovery_id: signature.recovery_id,
+    };
+    
+    let tx = program
+        .request()
+        .signer(sol.payer.clone())
+        .accounts(SolanaRespondAccount {
+            responder: sol.payer.clone().try_pubkey().unwrap(),
+        })
+        .args(SolanaRespond {
+            request_ids,
+            signatures: vec![signature],
+        })
+        .send()
+        .await
+        .map_err(|err| {
+            tracing::error!(
+                sign_id = ?action.request.indexed.id,
+                error = ?err,
+                "failed to publish solana signature"
+            );
+            crate::metrics::SIGNATURE_PUBLISH_FAILURES
+                .with_label_values(&[Chain::Solana.as_str(), "solana"])
+                .inc();
+        })?;
+
+    tracing::info!(
+        sign_id = ?action.request.indexed.id,
+        tx_hash = ?tx,
+        elapsed = ?timestamp.elapsed(),
+        "published solana signature successfully"
+    );
+    
+    crate::metrics::NUM_SIGN_SUCCESS
+        .with_label_values(&[Chain::Solana.as_str(), "solana"])
+        .inc();
+    crate::metrics::SIGN_TOTAL_LATENCY
+        .with_label_values(&[Chain::Solana.as_str(), "solana"])
+        .observe(action.request.indexed.timestamp.elapsed().as_secs_f64());
+    crate::metrics::SIGN_RESPOND_LATENCY
+        .with_label_values(&[Chain::Solana.as_str(), "solana"])
+        .observe(timestamp.elapsed().as_secs_f64());
+    
+    if action.request.indexed.timestamp.elapsed().as_secs() <= 30 {
+        crate::metrics::NUM_SIGN_SUCCESS_30S
+            .with_label_values(&[Chain::Solana.as_str(), "solana"])
```

Comment:
In all 4 calls to `with_label_values`, shouldn't the second label be the node account id?

Reply:
Good catch